csharp_dotnetcore/2.EchoBot-With-Counter/Startup.cs

```diff
@@ -48,7 +50,10 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddBot<EchoWithCounterBot>(options =>
             {
-                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+                // Load the connected services from .bot file.
+                var botConfig = BotConfiguration.Load(@".\EchoBotWithCounter.bot");
+                var endpointService = (EndpointService)botConfig.Services.First(s => s.Type == "endpoint");
```

Comment:
.FirstOrDefault(). First will throw if no endpoints exists, which probably isn't what's desired here.

Reply:
... especially as I see your have ? operators on the endpoint later, indicating you're expecting and good with null values coming back.